File: app-mvi/src/main/java/com/hannesdorfmann/mosby3/conductor/sample/Interactor.java
package com.hannesdorfmann.mosby3.conductor.sample;

import java.util.concurrent.TimeUnit;

import io.reactivex.Observable;

/**
 * Created by Michael on 12.02.2017.
 */

public class Interactor {

    public static Observable<String> loadData()
    {
        return Observable.just("Data loaded!")
                .delay(2000, TimeUnit.MILLISECONDS);
    }
}

File: app-mvi/src/main/java/com/hannesdorfmann/mosby3/conductor/sample/Interactor.kt
package com.hannesdorfmann.mosby3.conductor.sample

import io.reactivex.Observable
import io.reactivex.schedulers.Schedulers
import java.util.concurrent.TimeUnit

class Interactor {

    companion object {

        fun loadData(): Observable<PartialState> {

            return Observable.just("Data loaded!")
                    .delay(2000, TimeUnit.MILLISECONDS)
                    .map<PartialState> { PartialState.DataPartial(it) }
                    .startWith(PartialState.LoadingDataPartial)
                    .subscribeOn(Schedulers.io())

        }

    }


}